### PR TITLE
Scripts/ScarletMonastery; instance script cleanup

### PR DESCRIPTION
**Pull request by ccrs**
Functionally:
minor improvoements here and there, nothing fancy
headless might now be "playable" though (not having high hopes)

Technically:
streamline codestyle
streamline scripting methods & "best practices"
improve several fuckeries here and there, like abilities, phases or w/e

To test:
just see if everything works (every boss) lul

**Target branch(es):** 3.3.5

### CODE REVIEW

File: src/server/shared/Dynamic/LinkedReference/RefManager.h [hunk 1]
```diff
@@ -18,36 +18,38 @@
...
-        iterator begin() { return iterator(getFirst()); }
-        iterator end()   { return iterator(nullptr); }
+    iterator begin() { return iterator(getFirst()); }
+    iterator end() { return iterator(nullptr); }
 
-        virtual ~RefManager()
-        {
-            clearReferences();
-        }
+    const_iterator begin() const { return const_iterator(getFirst()); }
```
Comment (ccrs) [inline src/server/shared/Dynamic/LinkedReference/RefManager.h]: this is the actual change 💃

**Comment by ccrs** [comment]
@jackpoz @Keader give it a go please :)

**Comment by jackpoz** [comment]
Dynamic build:
>boss_high_inquisitor_fairbanks.obj : error LNK2019: unresolved external symbol "public: __cdecl DispelableAura::~DispelableAura(void)" (??1DispelableAura@@QEAA@XZ) referenced in function "public: void __cdecl std::list<class Unit *,class std::allocator<class Unit *> >::_Remove_if<class <lambda_49d84cb1bc2c217009cb01deb519a0f0> >(class <lambda_49d84cb1bc2c217009cb01deb519a0f0> &)" (??$_Remove_if@V<lambda_49d84cb1bc2c217009cb01deb519a0f0>@@@?$list@PEAVUnit@@V?$allocator@PEAVUnit@@@std@@@std@@QEAAXAEAV<lambda_49d84cb1bc2c217009cb01deb519a0f0>@@@Z) [C:\projects\trinitycore\build\src\server\scripts\scripts_easternkingdoms.vcxproj]
C:\projects\trinitycore\build\bin\RelWithDebInfo\scripts\scripts_easternkingdoms.dll : fatal error LNK1120: 1 unresolved externals [C:\projects\trinitycore\build\src\server\scripts\scripts_easternkingdoms.vcxproj]

**Comment by jackpoz** [comment]
This PR is hosted at https://tcubuntu.northeurope.cloudapp.azure.com/test (note the "/test" in the url, it requires to create a new account and update the realmlist compared to the usual https://tcubuntu.northeurope.cloudapp.azure.com ) .

**Comment by Aokromes** [comment]
https://i.imgur.com/IqOStdE.jpg xd

**Pull request merged**